### PR TITLE
fortune: 2.6.2 -> 2.10.0, use official tarball

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

https://github.com/shlomif/fortune-mod/blob/fortune-mod-2.10.0/fortune-mod/ChangeLog

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/fortune/default.nix [hunk 1]
```diff
@@ -1,35 +1,21 @@
...
   buildInputs = [ recode ];
 
-  preConfigure = ''
-    cp ${srcs.shlomifCommon} cmake/Shlomif_Common.cmake
-  '';
-
   postInstall = ''
-    mv $out/games/fortune $out/bin/fortune
-    rm -r $out/games
+    mv -v $out/games/fortune $out/bin/fortune
```
Comment (worldofpeace) [inline pkgs/tools/misc/fortune/default.nix]: I'm seeing the output directories like
```
local
│   └── share
│       └── games
```

Probably shouldn't have the `local`.

**Comment by dtzWill** [comment]
Rebased onto master (eep pname changes! :cat2:), fixed the `local` bit thanks!

File: pkgs/tools/misc/fortune/default.nix [hunk 2]
```diff
@@ -1,37 +1,38 @@
...
-  postInstall = ''
-    mv $out/games/fortune $out/bin/fortune
-    rm -r $out/games
-  '';
+  cmakeFlags = [
+    "-DLOCALDIR=${placeholder "out"}/share/fortunes"
+  ];
+
+  patches = [ (builtins.toFile "not-a-game.patch" ''
+    diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (worldofpeace) [inline pkgs/tools/misc/fortune/default.nix]: This is a pretty interesting way to produce a patch.

**Review by worldofpeace — APPROVED** [review]
Checked still builds, executed some stuff.

**Pull request merged**